Add DatePicker variant with no text field

amygl opened the pull request:
#493 DatePicker variant - No input text box
Added a DatePicker variant that will hide the text field, it will be opened via an external action (like a button click).

Added 3 optional props:
showDatePicker - When true the DatePicker's state will be updated to display the date picker.
hideTextField - When true text field will be hidden
onDismiss - callback for when the DatePicker is closed

msftclas commented:
Hi __@amygl__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftclas commented:
__@amygl__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

dzearing commented:
A general comment would be to split the component into smaller ones, instead of providing a "no input box" flag. That might be more elegant.

amygl commented:
I've refactored the code so that there is a separate Calendar component that is used by the DatePicker. This component can then be hooked up to a button or text field or other control.

dzearing commented:
Thanks Amy! I will deploy and play with it over the weekend when i get a chance.

MLoughry commented:
It seems that focus isn't being placed in the date picker when it opens, which is a regression from current behavior.

MLoughry commented:
You should add an example (or modify an existing one) to show off the `isMonthPickerVisible` option.

MLoughry commented:
Unrelated to your change, so feel free to ignore: we're not using `Layer` or `Callout` or similar logic for the Calendar, so it's getting clipped in the example pages.

If you're willing to do that work, it'd be awesome!

amygl commented:
I have fixed the focus regression and added an example to the Calendar demo page to show when the month picker is not visible.

MLoughry commented:
The general practice has been to conditional setting focus on actions where we explicitly expect focus to be set. Otherwise, a change in the props passed by the parent causes the Calendar to rerender and steal focus, even if that's not desired (for example, changing the month using the arrows in the day picker steals focus before the user can scroll further through the months). 

In the calendar case, I think it should set focus on update in the case of 'go to today' or selecting a month in the month picker. On mount, there should be a property dictating whether to set focus (which can be true by default).

amygl commented:
I've updated DatePicker to host Calendar in a callout, fixed up the focus issue, and added a calendar test.
The calendar test only tests rendering, I ran out of time to add a functionality test.

dzearing commented:
Addresses https://github.com/OfficeDev/office-ui-fabric-react/issues/331

dzearing commented on packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx:
```diff
@@ -0,0 +1,178 @@
+    selectedDate?: Date;
+}
+
+export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> {
```
nit: we use 2 spaces.
